app/src/main/java/org/wikipedia/gallery/GalleryViewState.kt

```diff
@@ -0,0 +1,13 @@
+package org.wikipedia.gallery
+
+/**Created by
+Author: Ankush Bose
+Date: 19,May,2021
+ **/
+
+sealed class GalleryViewState<out T> {
```

Comment:
If you see `Resource` there is no loading state,
Also there is no error part used in `LanguagesListViewModel.kt `.

However I can see .Error usage on `TalkReplyViewModel.kt`.

Data class is meant as container for values, where as open class is different concept. Additionally in `GalleryViewState` there is a base state called `InitialState` which is used as default value when mutable live data is initialized.

I am open to know more what do you think on my thought. Kindly educate me if I am missing something. @cooltey

Reply:
Since there's a similar class doing this (without loading and initial state handling), would it be possible to extend the `Resouce` and add `InitialState` and `Loading` in this class?